File: c++/Memory.cpp
#include "Hardware.cpp"

class Memory: public Hardware{
	private:
	/*atribut private*/
		string frequency;
		string memorySize;
		string supportsCuda;

	public:
		/*constructor*/
		Memory(){
			this->frequency = this->memorySize = this->supportsCuda = "";
		}

		/*getter/setter*/
		void setfrequency(string frequency){
			this->frequency = frequency;
		}
		string getfrequency(){
			return this->frequency;
		}
		void setmemorySize(string memorySize){
			this->memorySize = memorySize;
		}
		string getmemorySize(){
			return this->memorySize;
		}
		void setsupportsCuda(string supportsCuda){
			this->supportsCuda = supportsCuda;
		}
		string getsupportsCuda(){
			return this->supportsCuda;
		}
		/*destructor*/
		~Memory(){}
};

File: java/Memory.java
class Memory extends Hardware{
	// atribut private
    private String frequency;
	private String memorySize;
	private String supportsCuda;

	// setter/getter
	public void setfrequency(String frequency) {
		this.frequency = frequency;
	}
	public String getfrequency() {
		return frequency;
	}
	public void setmemorySize(String memorySize) {
		this.memorySize = memorySize;
	}
	public String getmemorySize() {
		return memorySize;
	}
	public void setsupportsCuda(String supportsCuda) {
		this.supportsCuda = supportsCuda;
	}
	public String getsupportsCuda() {
		return supportsCuda;
	}
}